src/programs/RelationalPrograms.jl

```diff
@@ -194,19 +200,26 @@ function parse_relation_context(context)
   vars = map(terms) do term
     @match term begin
       Expr(:(::), var::Symbol, type::Symbol) => (var => type)
+      Expr(:(::), var::Symbol, type::Int) => (var => type)
+      Expr(:(::), var::Symbol, type::Expr) => (var => type)
       var::Symbol => var
-      _ => error("Invalid syntax in term $expr of context")
+      _ => error("Invalid syntax in term $term of context")
     end
   end
   if vars isa AbstractVector{Symbol}
     (vars, nothing)
   elseif vars isa AbstractVector{Pair{Symbol,Symbol}}
     (first.(vars), last.(vars))
+  elseif vars isa AbstractVector{Pair{Symbol, Int}}
+    (first.(vars), last.(vars))
+  elseif vars isa AbstractVector{Pair{Symbol, Expr}}
+    (first.(vars), last.(vars))
   else
-    error("Context $context mixes typed and untyped variables")
+    error("Context $context mixes variable types")
```

Comment:
Only risk I see here is that when you have expressions, you might want to treat a symbol as the expression containing that symbol. Like how you might want to treat an Int as the list containing only that Int in some contexts. So there might be a valid input that comes out as `AbstractVector{Pair{Symbol, Union{Symbol,Expr}}}`